### ISSUE TITLE
Check for correctly divisible batch size

### ISSUE THREAD

**Issue by runame** [mlcommons/algorithmic-efficiency#119]
## Description

We should probably throw an informative error if
1) the training batch size is not divisible by the number of GPUs which are used,
2) the evaluation batch size is chosen such that the size of the remainder is not divisible by the number of GPUs which are used.

**Comment by pomonam** [comment]
Have we decided what behaviour we want to have for both cases? I think that we have two available options (please also let me know if there are other options). 
1. Raise Exceptions (e.g., `TrainBatchNotDivisibleError` and `EvalBatchNotDivisibleError`) for case 1 and 2, respectively. I assume that this should be easier for us (in terms of code change).
2. Pad the training and evaluation data given the batch size and number of GPUs in use. We can create dummy data so that conditions 1 and 2 are satisfied and assign weights 0 on these data points. Since we already have optional parameter masking, we can utilize the existing APIs. 

Please let me know what you think about these two potential solutions. If we settle on one decision, I can do the implementation.

**Comment by znado** [comment]
I definitely would prefer your option 2. Like you said, we can use `mask_batch` to tell submitters to ignore padded examples.

**Comment by runame** [comment]
I think you can just use the quick fix I mentioned in #204 for this, since it will only _very_ mildly affect the eval time and metrics on the test split and nothing else.

**Comment by znado** [comment]
Sorry, I missed that in the original comment. I agree, for training we should just assert that `batch size % N_GPUS == 0` and raise an error if not. Even if a submitter wanted to choose a very specific batch size, they should be able to round it to some factor of `N_GPUS` (which is 8 in our case). I think we should only pad the remainder batch for evals.

**Comment by runame** [comment]
@znado Should we pad to the closest multiple of `N_GPUS` or to the `global_eval_batch_size`? For PyTorch it doesn't really matter, but I think for Jax the latter option is better so jitted code doesn't get recompiled?
